packages/ra-core/src/dataProvider/useUpdate.ts

```diff
@@ -357,41 +289,43 @@ export const useUpdate = <RecordType extends Record = Record>(
         // except we do it in a mutate wrapper instead of the onMutate callback
         // to have access to success side effects
 
-        // Cancel any outgoing re-fetches (so they don't overwrite our optimistic update)
-        await queryClient.cancelQueries([
+        const previousRecord = queryClient.getQueryData<RecordType>([
             callTimeResource,
             'getOne',
             String(callTimeId),
         ]);
-        await queryClient.cancelQueries([callTimeResource, 'getList']);
 
-        // Snapshot the previous values
-        const previousGetOne: RecordType = queryClient.getQueryData([
-            callTimeResource,
-            'getOne',
-            String(callTimeId),
-        ]);
-        const previousGetList = queryClient.getQueriesData([
-            callTimeResource,
-            'getList',
-        ]);
-        const previousGetMany = queryClient.getQueriesData([
-            callTimeResource,
-            'getMany',
-        ]);
-        const previousGetManyReference = queryClient.getQueriesData([
-            callTimeResource,
-            'getManyReference',
-        ]);
-        rollbackData.current = {
-            previousGetOne,
-            previousGetList,
-            previousGetMany,
-            previousGetManyReference,
-        };
+        const queryKeys = [
+            [callTimeResource, 'getOne', String(callTimeId)],
+            [callTimeResource, 'getList'],
+            [callTimeResource, 'getMany'],
+            [callTimeResource, 'getManyReference'],
+        ];
 
-        // Optimistically update to the new value in getOne
-        await updateCache({
+        /**
+         * Snapshot the previous values via queryClient.getQueriesData()
+         *
+         * The snapshotData ref will contain an array of tuples [query key, associated data]
+         *
+         * @example
+         * [
+         *   [['posts', 'getOne', '1'], { id: 1, title: 'Hello' }],
+         *   [['posts', 'getList'], { data: [{ id: 1, title: 'Hello' }], total: 1 }],
+         *   [['posts', 'getMany'], [{ id: 1, title: 'Hello' }]],
+         * ]
+         *
+         * @see https://react-query.tanstack.com/reference/QueryClient#queryclientgetqueriesdata
+         */
+        snapshot.current = queryKeys.reduce(
+            (prev, curr) => prev.concat(queryClient.getQueriesData(curr)),
+            [] as Snapshot
+        );
+
+        // Cancel any outgoing re-fetches (so they don't overwrite our optimistic update)
+        await snapshot.current.map(([key]) => queryClient.cancelQueries(key));
```

Comment:
~~await~~ snapshot.current.map([key]) => queryClient.cancelQueries(key));

`await` has no effect here

Reply:
It was a `forEach` before, If not using the result, why change it to map?

Reply:
I'll use Promise.all() instead